src/constraints/util/operators.ts

```diff
@@ -38,3 +38,8 @@ export interface Comparator {
 	(a: number, b: number): boolean;
 	(a: bigint, b: bigint): boolean;
 }
+
+export interface UrlOptions {
+	allowedProtocols?: string[];
```

Comment:
Should require `:` at the end of all strings, make this a `StringProtocol` type and export it:
```suggestion
	allowedProtocols?: `${string}:`[];
```